sktime/forecasting/base/adapters/_pytorchforecasting.py

```diff
@@ -0,0 +1,445 @@
+        # store the target column names and index names (probably [None])
+        # will be renamed !
+        self._target_name = _y.columns[-1]
+        self._index_names = _y.index.names
+        self._index_len = len(self._index_names)
+        # store X, y column names (probably None or not str type)
+        # will be renamed !
+        if X is not None:
+            self._X_columns = X.columns.tolist()
+        # convert data to pytorch-forecasting datasets
+        training, validation = self._Xy_to_dataset(
+            _X, _y, self._dataset_params, self._max_prediction_length
+        )
+        # instantiate forecaster and trainer
+        self._forecaster, self._trainer = self._instantiate_model(training)
+        # convert dataset to dataloader
+        self._train_to_dataloader_params = {"train": True}
+        self._train_to_dataloader_params.update(
+            _none_check(self.train_to_dataloader_params, {})
+        )
+        self._validation_to_dataloader_params = {"train": False}
+        self._validation_to_dataloader_params.update(
+            _none_check(self.validation_to_dataloader_params, {})
+        )
+        # call the fit function of the pytorch-forecasting model
+        self._trainer.fit(
+            self._forecaster,
+            train_dataloaders=training.to_dataloader(
+                **self._train_to_dataloader_params
+            ),
+            val_dataloaders=validation.to_dataloader(
+                **self._validation_to_dataloader_params
+            ),
+        )
+        return self
+
+    def _predict(
+        self: "_PytorchForecastingAdapter",
+        fh: typing.Optional[ForecastingHorizon],
+        X: typing.Optional[pandas.DataFrame],
+        y: typing.Optional[pandas.DataFrame],
+    ) -> pandas.Series:
+        """Forecast time series at future horizon.
+
+        private _predict containing the core logic, called from predict
+
+        State required:
+            Requires state to be "fitted".
+
+        Accesses in self:
+            Fitted model attributes ending in "_"
+            self.cutoff
+
+        Parameters
+        ----------
+        fh : guaranteed to be ForecastingHorizon or None, optional (default=None)
+            The forecasting horizon with the steps ahead to to predict.
+        X : sktime time series object, optional (default=None)
+            guaranteed to be of an mtype in self.get_tag("X_inner_mtype")
+            Exogeneous time series for the forecast
+        y : sktime time series object, optional (default=None)
+            Historical values of the time series that should be predicted.
+
+        Returns
+        -------
+        y_pred : sktime time series object
+            guaranteed to have a single column/variable
+            Point predictions
+        """
+        if y is None:
+            y = deepcopy(self._y)
+        if X is None:
+            X = deepcopy(self._X)
+        if X is not None and not self._global_forecasting:
```

Comment:
I haven't defined it yet. I will add it in the docstring of `predict` function. Should we have a new page to explain it?

Reply:
This would be cool if you can add this in the docstring of the predict function of the base class. 

We should also create an issue that introduce an example in one of our tutorial notebooks. :)

Reply:
Docstring is added to `predict` function both of `_PytorchForecastingAdapter` and `BaseGlobalForecaster` now.

Issue: [DOC] Global Forecast Example #6575